### ISSUE TITLE
Rewrite remaining Java classes in Kotlin

### ISSUE THREAD

**Issue by MSF-Jarvis** [MSF-Jarvis/wireguard-kt#50]

**Comment by theLimitBreaker** [comment]
I would like to take this issue .

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Awesome!

```
msfjarvis@ryzenbox ~/git-repos/wireguard-kt (master) $ find app/src/main/java/com/wireguard/ -type f -name '*.java'
app/src/main/java/com/wireguard/crypto/KeyEncoding.java
app/src/main/java/com/wireguard/crypto/Curve25519.java
app/src/main/java/com/wireguard/crypto/Keypair.java
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/InetEndpoint.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/util/Keyed.java
app/src/main/java/com/wireguard/util/SortedKeyedList.java
app/src/main/java/com/wireguard/util/KeyedList.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/SlashDrawable.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/KotlinCompanions.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/databinding/ObservableKeyedRecyclerViewAdapter.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java
```

These are the current Java classes in the project. The crypto code is external so I'd like to keep it close to the respective sources so that should stay in Java. Converting the `util` package and all the `Observable*` classes that implement the interfaces from the `util` package is the current heavyweight task in terms of full Kotlin migration. Or you can take a stab at the `AsyncWorker` and `RootShell` classes (feel free to replace the entire infrastructure around these if you feel up to it, I'll be glad).

Summing up, the major buckets of Kotlin rewrites here are as below:

- [ ] Rewrite `util` package and all `Observable*` classes
- [ ] Rewrite `AsyncWorker` and `RootShell` classes
- [ ] Rewrite `GoBackend` class

If you take up the final task, I can push a mostly finished patch to a separate branch for you to reference or work off of.

**Comment by MSF-Jarvis** [comment]
Super massive connection fart there, sorry for the spam :(

**Comment by theLimitBreaker** [comment]
Sure . I'll do it

**Comment by MSF-Jarvis** [comment]
The entire checklist? \o/

**Comment by MSF-Jarvis** [comment]
I've gone ahead and rebased the `gobackend-kt` and `rootshell-rewrite-v2` branches for your perusal. Good luck!

**Comment by theLimitBreaker** [comment]
Some file is missing  Cmake produces this:

```
CMake Error at CMakeLists.txt:11 (add_executable):
  Cannot find source file:

    wireguard/src/tools/wg-quick/android.c
```

**Comment by MSF-Jarvis** [comment]
@theLimitBreaker you need to sync down all submodules. Try `git submodule update --init --recursive`.

**Comment by theLimitBreaker** [comment]
yeah I checked the wireguard-mirror it said the same to add sub modules thanks

**Comment by theLimitBreaker** [comment]
Still One Build Failing 

`> Build command failed.
  Error while executing process /home/thelimitbreaker/Android/Sdk/cmake/3.6.4111459/bin/cmake with arguments {--build /home/thelimitbreaker/forked_cloned/wireguard-kt/app/.externalNativeBuild/cmake/debug/x86_64 --target libwg.so}
`

**Comment by theLimitBreaker** [comment]
gradle Sync failed .  
workstation : ubuntu 18.04

gradle plugin : 3.3.0-alpha12 . 
gradle-wrapper.properties

```
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-all.zip
```

`Unsupported method: NativeArtifact.getSourceFolders().
The version of Gradle you connect to does not support that method.
To resolve the problem you can change/upgrade the target version of Gradle you connect to.
Alternatively, you can ignore this exception and read other information from the model`

**Comment by MSF-Jarvis** [comment]
If you have the Android NDK installed, uninstall it and use `sdkmanager` to install ndk-bundle instead.

**Comment by MSF-Jarvis** [comment]
Anything I can help with @theLimitBreaker? I don't see any progress on your fork yet.

**Comment by theLimitBreaker** [comment]
There were some issues with my machine  @MSF-Jarvis  Now it's fine.

**Comment by theLimitBreaker** [comment]
@MSF-Jarvis  Should I make PR after converting some 2-3 files into kotlin ?

**Comment by MSF-Jarvis** [comment]
@theLimitBreaker yes that'd be optimal.

**Comment by MSF-Jarvis** [comment]
Got rid of some of the java classes with c8efdd5ae9be

Remaining now

```
app/src/main/java/com/wireguard/config/Config.java
app/src/main/java/com/wireguard/config/Peer.java
app/src/main/java/com/wireguard/android/widget/KeyInputFilter.java
app/src/main/java/com/wireguard/android/widget/NameInputFilter.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/AsyncWorker.java
app/src/main/java/com/wireguard/android/util/ObservableSortedKeyedArrayList.java
app/src/main/java/com/wireguard/android/util/ObservableKeyedList.java
app/src/main/java/com/wireguard/android/util/RootShell.java
app/src/main/java/com/wireguard/android/backend/GoBackend.java```

**Comment by MSF-Jarvis** [comment]
Config.java [is gone](https://github.com/MSF-Jarvis/wireguard-kt/commit/fadb71de6dcf74da17de166fe1530ee690e753b8) as well

**Comment by MSF-Jarvis** [comment]
`app/src/main/java/com/wireguard/android/backend/GoBackend.java` is now in kotlin pending merge of #86

**Comment by MSF-Jarvis** [comment]
Done for now, will revisit later.
